### ISSUE TITLE
Multinomial logistic regression

### ISSUE THREAD

**Issue by leo60228** [rust-ml/linfa#157]
It's worth noting here that I'm not sure this is actually what I'm looking for. A core component of a project I'd like to work on is effectively a multinomial classification problem. However, the project as a whole is much more complex than this, and I actually find the classification itself to be one of the least interesting parts. Based on prior art with this problem, I know that scikit-learn's logistic regression gives good results with my dataset. I was hoping to be able to easily do this in Rust.

**Comment by leo60228** [comment]
Comparing linfa and sklearn, it looks like it should be relatively easy to port over multinomial support. However, I'm somewhat worried about implementing this when I don't understand the math at all...

**Comment by YuhanLiin** [comment]
It should be relatively simple to extend/generalize the existing binomial regression in `linfa-logistic`. You'd need to make the weight vector into a weight matrix (n_features x n_targets) and extending the target label processing to support more than 2 distinct labels. It'll specifically involve extending the `argmin_params` module to multi-dimension arrays. I'd be interested in implementing myself this actually.

**Comment by YuhanLiin** [comment]
Implemented by #159
